Lib/threading.py

```diff
@@ -1515,22 +1515,19 @@ def _shutdown():
     """
     Wait until the Python thread state of all non-daemon threads get deleted.
     """
-    # Obscure: other threads may be waiting to join _main_thread.  That's
-    # dubious, but some code does it. We can't wait for it to be marked as done
-    # normally - that won't happen until the interpreter is nearly dead. So
-    # mark it done here.
-    if _main_thread._handle.is_done() and _is_main_interpreter():
```

Comment:
This early return logic started in https://github.com/python/cpython/commit/ee84a608587b930176d37303afae8a4358e15990 but no longer seems relevant in 3.13 given the other refactoring in threading.py cleaning up the locking mess.